application/modules/front/controllers/CollectionController.php

```diff
@@ -173,15 +173,19 @@ public function actionDispbook($collectionName, $ourBookID, $hadithNumbers = NUL
             if ($this->_book->status > 3 and count($pairs) === 1) { // If it's a single-hadith view page
                 $urn = $this->_entries[0][$pairs[0][0]]->englishURN;
                 $nextURN = $this->util->getNextURNInCollection($urn);
-                $previousURN = $this->util->getPreviousURNInCollection($urn);
+				$previousURN = $this->util->getPreviousURNInCollection($urn);
+				$permalinkCanonical = $this->util->get_permalink($this->_entries[1][$pairs[0][1]]->arabicURN, "arabic");
```

Comment:
You should just be able to use `$pairs[0][1]->permalink` here instead of calling `get_permalink(...)`. Can you change and verify it doesn't break anything? In fact it should be possible to use this everywhere instead of `get_permalink(...)`.

Reply:
This condition is already satisfied above. See line